Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -191,24 +243,62 @@ internal class ResponseAccumulator: HTTPClientResponseDelegate {
     }
 }
 
-/// This delegate is strongly held by the HTTPTaskHandler
-/// for the duration of the HTTPRequest processing and will be
-/// released together with the HTTPTaskHandler when channel is closed
+/// HTTPClientResponseDelegate allows to receive notification about request processing and to control how reponse parts are processed.
```

Comment:
Backticks around type names.

I think this needs a more user-centric explanation, something like: "You can implement this protocol if you need fine control over an HTTP request/response, for example if you want to inspect the response headers before deciding whether to accept a response body, or if you want to stream your request body. Pass an instance of your conforming class to the `HTTPClient.execute()` method and this package will call each delegate method appropriately as the request takes place."

Just a thought.

Reply:
Great idea, added!